### PR TITLE
Updated template to work with API Version 2020-06-01

### PR DESCRIPTION
**Pull request by StefanIvemo**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

* Updated API Version to 2020-06-01 for all Microsoft.Network/* resources
* Removed VNet Connection from Virtual Hub Resource and created a separate child resource to work with API Version 2020-06-01
* Updated Firewall Policy to work with API Version 2020-06-01
* Replaced Virtual Hub Route Table with Virtual Hub Hub Route Table
* Removed Route Table BranchRouteTable that wasn't used by the resources in the example

### CODE REVIEW

**Comment by bmoore-msft** [comment]
@vhorne - check this PR for updates that may be needed to documentation that references this sample.  [this is an automated message]

File: fwm-docs-qs/azuredeploy.json [hunk 1]
```diff
@@ -32,95 +32,115 @@
...
+            ],
+            "ids": [
+              {
+                "id": "[resourceId('Microsoft.Network/virtualHubs/hubRouteTables', 'Hub-01', 'RT_VNet')]"
+              }
+            ]
+          }
+        }
+      },
+      "dependsOn": [
```
Comment (bmoore-msft) [inline fwm-docs-qs/azuredeploy.json:87]: See: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#sort-order-of-properties for the proper sort order of properties

i.e. move this above properties

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: fwm-docs-qs/azuredeploy.json (hunk 1 above)
Comment (StefanIvemo) [reply fwm-docs-qs/azuredeploy.json]: Moved depensOn above properties and pushed a new commit

**Comment by bmoore-msft** [comment]
@vhorne - check this PR for updates that may be needed to documentation that references this sample.

File: fwm-docs-qs/azuredeploy.json [hunk 2]
```diff
@@ -32,95 +32,115 @@
...
+      "apiVersion": "2020-06-01",
+      "name": "Hub-01/hub-spoke",
+      "dependsOn": [
+        "[resourceId('Microsoft.Network/azureFirewalls', 'AzfwTest')]",
+        "[resourceId('Microsoft.Network/virtualHubs', 'Hub-01')]",
+        "[resourceId('Microsoft.Network/virtualNetworks', 'Spoke-01')]",
+        "[resourceId('Microsoft.Network/virtualHubs/hubRouteTables', 'Hub-01', 'RT_VNet')]"
+      ],
+      "properties": {
+        "remoteVirtualNetwork": {
```
Comment (bmoore-msft) [inline fwm-docs-qs/azuredeploy.json:71]: format the JSON (e.g. SHIFT+ALT+F in VS Code)

**Review by bmoore-msft — CHANGES_REQUESTED** [review]
Also see the pipeline results for changes that need to be made

File: fwm-docs-qs/azuredeploy.json (hunk 2 above)
Comment (StefanIvemo) [reply fwm-docs-qs/azuredeploy.json:71]: JSON is formatted already using SHIFT+ALT+F.

File: fwm-docs-qs/azuredeploy.json [hunk 3]
```diff
@@ -32,95 +32,115 @@
...
       }
     },
     {
-      "type": "Microsoft.Network/firewallPolicies/ruleGroups",
-      "apiVersion": "2020-04-01",
+      "type": "Microsoft.Network/firewallPolicies/ruleCollectionGroups",
+      "apiVersion": "2020-06-01",
       "name": "Policy-01/DefaultApplicationRuleCollectionGroup",
-      "location": "[parameters('location')]",
+      "location": "westeurope",
```
Comment (bmoore-msft) [inline fwm-docs-qs/azuredeploy.json:107]: this must be a parameter

File: fwm-docs-qs/azuredeploy.json [hunk 4]
```diff
@@ -32,95 +32,115 @@
...
-            "ruleType": "FirewallPolicyFilterRule",
             "action": {
               "type": "Allow"
             },
-            "ruleConditions": [
+            "rules": [
               {
+                "ruleType": "ApplicationRule",
                 "name": "Allow-msft",
+                "description": "",
```
Comment (bmoore-msft) [inline fwm-docs-qs/azuredeploy.json:125]: remove all empty/null properties

File: fwm-docs-qs/azuredeploy.json (hunk 3 above)
Comment (bmoore-msft) [inline fwm-docs-qs/azuredeploy.json:107]: this must be a parameter

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

**Comment by bmoore-msft** [comment]
@vhorne - check this PR for updates that may be needed to documentation that references this sample.

**Comment by bmoore-msft** [comment]
@vhorne - check this PR for updates that may be needed to documentation that references this sample.

**Pull request merged**